tests/functionality/toml_utils.hpp

```diff
@@ -0,0 +1,230 @@
+#pragma once
```

Comment:
Erroneously named (toml_utils.hpp), as we already have something called `toml_utils.h`

Reply:
toml_utils.hpp doesn't appear to be used anywhere and deleting it doesn't appear to have any effect on the test suite. I also did a diff on toml_utils.hpp and toml_utils.h and toml_utils.h appears to have more recent updates so I think it is safe to get rid of toml_utils.hpp. I will do this on the func-test-mpi branch.

Reply:
The first issue has been close and the second and third merge PRs have been merged. The fourth PR is duplicated in Github by this PR so the only outstanding issue is 284 in Gitlab. I will copy this issue to Github.

Reply:
Okay so is there a specific change that this PR is addressing, or does more work need to be done in order to address SCOPFLOW/SOPFLOW parallel tests?

Reply:
Thanks for cleaning this up! Next steps then are to implement the fixes in `selfcheck.cpp` for both SOPFLOW and SCOPFLOW, and then we can merge.